swaggen/src/main/java/generator/PathGenerator.java

```diff
@@ -54,9 +54,59 @@
 	 * @param annotation the annotation
 	 * @return the path
 	 */
-	private static Map<RequestMethod, Endpoint> generatePathFromAnnotation(SwaggerGen annotation) {
-		Map<RequestMethod, Endpoint> endpointMap = new HashMap<>();
-		endpointMap.put(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
-		return endpointMap;
+	private static SwaggerEndpoint generatePathFromAnnotation(SwaggerGen annotation) {
+		SwaggerEndpoint endpoint = new SwaggerEndpoint();
+		endpoint.addEndpoint(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
+		return endpoint;
+	}
+
+	/**
+	 * Checks if URL is already in the existing map, and if it is, appends the swagger
+	 * endpoints to the URL
+	 * Case: multiple classes with the same endpoint
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 * @param classPaths Map of new Swagger Endpoints in the class
+	 */
+	private static void checkClassEndpoints(Map<PathURL, SwaggerEndpoint> existingPaths, Map<PathURL, SwaggerEndpoint> classPaths) {
+		if (existingPaths.isEmpty()) {
+			existingPaths.putAll(classPaths);
+			return;
+		}
+
+		for(PathURL pathKey: existingPaths.keySet()) {

```

Comment:
This loop still isn't quite right and that may be partially my fault. Imagine we have 2 maps:
`{ a: [A1], b: [B1], d: [D1] }` and `{ a: [A2], b: [B2], c: [C2] }`

What we would want is: `{ a: [A1, A2], b: [B1, B2], c: [C2] }`
What we would get is: `{ a: [A2], b: [B2], c: [C2], d: [D1] }` 

My original comment mixed up which keys we should loop through. You should loop through the keys of the map you are not writing too. 

So:
```
for (PathUri pathKey : classPaths.keysSet()) {
    if(existingPaths.containsKey(pathKey) {
        // Extend the exsisting list
        existingPaths.get(pathKey).addEndpoint(classPaths.get(pathKey));
    else {
         // Add this key (don't putAll, or you'll overwrite all common keys)
         existingPaths.Put(pathKey, classPaths.get(pathKey));
}

Sorry for the confusion.